utilities/cloudharness_utilities/deployment-configuration/helm/files/db-mongo-backup.sh

```diff
@@ -0,0 +1,35 @@
+/bin/bash << 'EOF'
+KEEP_DAYS=${BACKUP_KEEP_DAYS}
+KEEP_WEEKS=`expr $(((${BACKUP_KEEP_WEEKS} * 7) + 1))`
+KEEP_MONTHS=`expr $(((${BACKUP_KEEP_MONTHS} * 31) + 1))`
+
+mkdir -p "${BACKUP_DIR}/daily/" "${BACKUP_DIR}/weekly/" "${BACKUP_DIR}/monthly/"
+DFILE="${BACKUP_DIR}/daily/`date +%Y-%m-%d-%H%M%S`${BACKUP_SUFFIX}"
+WFILE="${BACKUP_DIR}/weekly/`date +%G-%V`${BACKUP_SUFFIX}"
+MFILE="${BACKUP_DIR}/monthly/`date +%Y-%m`${BACKUP_SUFFIX}"
+
+# Dump mongo database
+/usr/bin/mongodump -h $DB_HOST -u $DB_USER -p $DB_PASS --archive=$DFILE --gzip
+
+# Use hardlink instead of copy to save space
+if [ -d "${DFILE}" ]; then
+    WFILENEW="${WFILE}-new"
+    MFILENEW="${MFILE}-new"
+    rm -rf "${WFILENEW}" "${MFILENEW}"
+    mkdir "${WFILENEW}" "${MFILENEW}"
+    ln -f "${DFILE}/"* "${WFILENEW}/"
+    ln -f "${DFILE}/"* "${MFILENEW}/"
+    rm -rf "${WFILE}" "${MFILE}"
+    mv -v "${WFILENEW}" "${WFILE}"
+    mv -v "${MFILENEW}" "${MFILE}"
+else
+    ln -vf "${DFILE}" "${WFILE}"
+    ln -vf "${DFILE}" "${MFILE}"
+fi
+
+# Clean old files
```

Comment:
I would do cleanup before create so we have more diskspace free before dumping

Reply:
sounds good